File: Project/travel_application/lib/models/trip.dart
import 'package:json_annotation/json_annotation.dart';
part 'trip.g.dart';

@JsonSerializable()
class Trip{
  final int id;
  final String title;
  final String destination;
  final int numberOfDays;
  final int numberOfPeople;
  final String imageUrl;
  final String description;
  final String creator;

  Trip({
  required this.id,
  required this.title, 
  required this.destination, 
  required this.numberOfDays, 
  required this.numberOfPeople, 
  required this.imageUrl,
  required this.description,
  required this.creator,
  });

  factory Trip.fromJson(Map<String, dynamic> json) => _$TripFromJson(json); 
  Map<String, dynamic> toJson() => _$TripToJson(this); 
}

File: Project/TripsServer/src/models/trip.ts
export class Trip {
    id: number;
    title: string;
    destination: string;
    numberOfDays: number;
    numberOfPeople: number;
    imageUrl: string;
    description: string;
    creator: string;

    constructor(
        id: number, title: string, destination: string, numberOfDays: number,
        numberOfPeople: number, imageUrl: string, description: string, creator:string
    ) {
        this.id = id;
        this.title = title;
        this.destination = destination;
        this.numberOfDays = numberOfDays;
        this.numberOfPeople = numberOfPeople;
        this.imageUrl = imageUrl;
        this.description = description;
        this.creator = creator;
    }
}
